src/extension/provider/launcher.ts

```diff
@@ -0,0 +1,124 @@
+import glob from 'glob'
+import { join, basename, dirname, resolve } from 'path'
+import * as vscode from 'vscode'
+
+interface IRunmeFileProps {
+  tooltip: string
+  lightIcon: string
+  darkIcon: string
+  collapsibleState: vscode.TreeItemCollapsibleState
+  onSelectedCommand?: vscode.Command
+  contextValue: string
+}
+
+export class RunmeFile extends vscode.TreeItem {
+  constructor(
+    public readonly label: string,
+    { collapsibleState, tooltip, onSelectedCommand, lightIcon, darkIcon, contextValue }: IRunmeFileProps
+  ) {
+    super(label, collapsibleState)
+    const assetsPath = join(__filename, '..', '..', 'assets')
+    this.tooltip = tooltip
+    this.label = label
+    this.command = onSelectedCommand
+    this.contextValue = contextValue
+    this.iconPath = {
+      light: join(assetsPath, lightIcon),
+      dark: join(assetsPath, darkIcon),
+    }
+  }
+}
+
+export class RunmeLauncherProvider implements vscode.TreeDataProvider<RunmeFile> {
+  private filesTree: Map<string, any>
+  private workspaceRoot: string
+  constructor() {
+    const rootPath =
+      vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0
+        ? vscode.workspace.workspaceFolders[0].uri.fsPath
+        : undefined
+    this.workspaceRoot = rootPath || ''
+    this.filesTree = new Map()
+  }
+
+  private _onDidChangeTreeData: vscode.EventEmitter<RunmeFile | undefined | void> = new vscode.EventEmitter<
+    RunmeFile | undefined | void
+  >()
+
+  readonly onDidChangeTreeData: vscode.Event<RunmeFile | undefined | void> = this._onDidChangeTreeData.event
+  getTreeItem(element: RunmeFile): vscode.TreeItem | Thenable<vscode.TreeItem> {
+    return element
+  }
+  getChildren(element?: RunmeFile | undefined): Thenable<RunmeFile[]> {
+    if (!this.workspaceRoot) {
+      return Promise.resolve([])
+    }
+
+    if (!element) {
+      return new Promise((resolve: (value: RunmeFile[]) => void) => {
+        this.getRunmeFilesFromWorkspace(resolve)
+      })
+    }
+
+    const { files, folderPath } = this.filesTree.get(element.label)
+    const folderMarkdownItems: RunmeFile[] = []
+    for (const file of files) {
+      folderMarkdownItems.push(
+        new RunmeFile(file, {
+          collapsibleState: vscode.TreeItemCollapsibleState.None,
+          tooltip: 'Click to open markdown file',
+          lightIcon: 'icon.gif',
+          darkIcon: 'icon.gif',
+          contextValue: 'markdown-file',
+          onSelectedCommand: {
+            arguments: [{ file, folderPath }],
+            command: 'runme.openRunmeFile',
+            title: file,
+          },
+        })
+      )
+    }
+
+    return Promise.resolve(folderMarkdownItems)
+  }
+
+  refresh(): void {
+    this._onDidChangeTreeData.fire()
+  }
+
+  getRunmeFilesFromWorkspace(onComplete: (value: RunmeFile[]) => void): any {
+    glob('**/*.md', { cwd: this.workspaceRoot, ignore: 'node_modules/**', absolute: true }, (err, files) => {
```

Comment:
@degrammer you do want to use the VS Code API's way to glob/traverse the FS. The reason is that vscode can run against virtual filesystems when attached to a SSH host or directly attached to a Github Repository. It's really only guaranteed that local FS is available when VS Code is run as Desktop app. It's something to get used to building inside of VS Code.

![image](https://user-images.githubusercontent.com/250527/212417881-106c4249-b03a-4818-86e7-e24af5fe8775.png)
Can't deep link to the API hence screenshot: https://code.visualstudio.com/api/references/vscode-api#workspace